src/ts/fs.d.ts

```diff
@@ -0,0 +1,379 @@
+         *   If `mode` is not supplied, the default of `0o666` is used.
+         *   If `flag` is not supplied, the default of `'a'` is used.
+         */
+        appendFile(file: NjsStringLike, data: NjsStringLike | Buffer, options?: WriteFileOptions): Promise<void>;
+
+        /**
+         * Asynchronously creates a directory at the specified `path`.
+         *
+         * @since 0.4.2
+         * @param path A path to a file.
+         * @param options The file mode (or an object specifying the file mode). Defaults to `0o777`.
+         */
+        mkdir(path: NjsStringLike, options?: { mode?: number } | number): Promise<void>;
+
+        /**
+         * Asynchronously reads the contents of a directory at the specified `path`.
+         *
+         * @since 0.4.2
+         * @param path A path to a file.
+         * @param options A string that specifies encoding or an object optionally specifying
+         *   the following keys:
+         *   - `encoding` - `'utf8'` (default) or `'buffer'` (since 0.4.4)
+         *   - `withFileTypes` - if set to `true`, the files array will contain `fs.Dirent` objects; defaults to `false`.
+         */
+        readdir(path: NjsStringLike, options?: { encoding?: "utf8"; withFileTypes?: false; } | "utf8"): Promise<NjsStringLike[]>;
+        readdir(path: NjsStringLike, options: { encoding: "buffer"; withFileTypes?: false; } | "buffer"): Promise<Buffer[]>;
+        readdir(path: NjsStringLike, options: { encoding?: "utf8" | "buffer"; withFileTypes: true; }): Promise<Dirent[]>;
+
+        /**
+         * Asynchronously returns the contents of the file with provided `filename`.
+         * If an encoding is specified, a `string` is returned, otherwise, a `Buffer`.
+         *
+         * @param path A path to a file.
+         * @param options A string that specifies encoding or an object with the following optional keys:
+         *   - `encoding` - `'utf8'`, `'hex'`, `'base64'`, or `'base64url'` (the last three since 0.4.4).
+         *   - `flag` - file system flag, defaults to `r`.
+         */
+        readFile(path: NjsStringLike): Promise<Buffer>;
+        readFile(path: NjsStringLike, options?: { flag?: OpenMode; }): Promise<Buffer>;
+        readFile(path: NjsStringLike, options: { encoding?: FileEncoding; flag?: OpenMode; } | FileEncoding): Promise<NjsStringLike>;
+
+        /**
+         * Asynchronously computes the canonical pathname by resolving `.`, `..` and symbolic links using
+         * `realpath(3)`.
+         *
+         * @since 0.3.9
+         * @param path A path to a file.
+         * @param options The encoding (or an object specifying the encoding), used as the encoding of the result.
+         */
+        realpath(path: NjsStringLike, options?: { encoding?: "utf8" } | "utf8"): Promise<NjsStringLike>;
+        realpath(path: NjsStringLike, options: { encoding: "buffer" } | "buffer"): Promise<Buffer>;  // TODO: is this correct?
```

Comment:
Is this correct?

Reply:
realpath(path: NjsStringLike, options?: { encoding?: "utf8" } | "utf8"): Promise<string>;
realpath(path: NjsStringLike, options: { encoding: "buffer" } | "buffer"): Promise<Buffer>; // yes, this is correct